autocorpus/parse_xml.py

```diff
@@ -0,0 +1,1431 @@
+    if curren_title is not None:
+        target_title = soup2.find(
+            "title", string=f"{curren_title.text}"
+        )  # Find the exact title in the soup2 object
+
+        if target_title is not None:
+            # Find the hierarchy of parent titles for the current title
+            parent_titles = find_parent_titles(target_title)
+        else:
+            parent_titles = []
+
+        # If there are multiple parent titles, exclude the last one (likely redundant)
+        if len(parent_titles) > 1:
+            parent_titles = parent_titles[:-1]
+    else:
+        parent_titles = None  # If no current title, set parent titles to None
+
+    # Extract the content within the current section, specifically paragraphs (`<p>` tags)
+    content = BeautifulSoup("<sec" + str(section).split("<sec")[1]).find_all("p")
+
+    # If content is found, process each paragraph
+    if content:
+        for i in range(len(content)):
+            # Avoid adding duplicate or null content
+            if f"{content[i]}" not in text_list and content[i] is not None:
+                # If no parent titles are found, tag the content with 'Unknown'
+                if parent_titles is None:
+                    tag_title.append(["Unknown"])
+                else:
+                    # Otherwise, tag it with the identified parent titles
+                    tag_title.append(parent_titles)
+
+                # Similarly, handle subtitle tagging
+                if curren_title is None:
+                    tag_subtitle.append("Unknown")
+                else:
+                    tag_subtitle.append(f"{[curren_title.text]}")
+
+                # Add the processed content to the text list
+                text_list.append(f"{content[i]}")
+
+    # Recursively process any subsections within the current section
+    subsections = section.find_all("sec")
+    for subsection in subsections:
+        extract_section_content(subsection, soup2, ori_title)
+
+
+def find_parent_titles(title_element):
+    # Initialize an empty list to store parent titles
+    parent_titles = []
+
+    # Find the immediate parent <sec> element of the given title element
+    parent = title_element.find_parent(["sec"])
+
+    # Traverse up the hierarchy of <sec> elements
+    while parent:
+        # Look for a <title> within the current parent <sec>
+        title = parent.find("title")
+        if title:
+            # If a title is found, add it to the beginning of the parent_titles list
+            parent_titles.insert(0, title.text)
+
+        # Move up to the next parent <sec> in the hierarchy
+        parent = parent.find_parent(["sec"])
+
+    # Return the list of parent titles, ordered from topmost to immediate parent
+    return parent_titles
+
+
+if __name__ == "__main__":
+    ######### INPUT #########
+
+    ### A directory containing the path to the XML files to be processed, is a list of str
+    all_files = glob.glob("./xml_hackathon/*.xml")
```

Comment:
I know the plan was to commit this script verbatim for now, but I'm wondering if we could make this script a lot more useful by just allowing the user to pass in paths to one or more XML files as arguments, so people can do:

```sh
python -m autocorpus.parse_xml my_file.xml
```